File: 10161/main.py
import sys
import math

def main():
	for step in sys.stdin:
		step = int(step)
		if step == 0:
			break
		sqrt = int(math.sqrt(step))
		if (sqrt ** 2) != step:
			sqrt += 1
		m = (sqrt ** 2) - (sqrt - 1)
		if (sqrt % 2) == 0:
			if step > m:
				print(str(sqrt), str(sqrt - (step - m)))
			else:
				print(str(sqrt - (m - step)), str(sqrt))
		else:
			if step > m:
				print(str(sqrt - (step - m)), str(sqrt))
			else:
				print(str(sqrt), str(sqrt - (m - step)))

if __name__ == '__main__':
	main()

File: 10161/Main.java
import java.io.*;

public class Main {
	public static void main(String args[]) throws IOException {
		String line;
		StringBuilder answer = new StringBuilder();
		BufferedReader in = new BufferedReader(new InputStreamReader(System.in));

		while ((line = in.readLine()) != null) {
			int step;
			int sqrt;
			int m, x, y;
			step = Integer.parseInt(line);
			if (step == 0)
				break;
			sqrt = (int) Math.sqrt((double) step);
			if ((sqrt * sqrt) != step)
				++sqrt;
			m = (sqrt * sqrt) - (sqrt - 1);
			if ((sqrt % 2) == 0) {
				if (step > m) {
					answer.append(Integer.toString(sqrt) + " " + Integer.toString(sqrt - (step - m)) + "\n");
				} else {
					answer.append(Integer.toString(sqrt - (m - step)) + " " + Integer.toString(sqrt) + "\n");
				}
			} else {
				if (step > m) {
					answer.append(Integer.toString(sqrt - (step - m)) + " " + Integer.toString(sqrt) + "\n");
				} else {
					answer.append(Integer.toString(sqrt) + " " + Integer.toString(sqrt - (m - step)) + "\n");
				}
			}
		}
		System.out.print(answer);
	}
}
